### PR TITLE
feat: init flowy document 2

### PR DESCRIPTION
**Pull request by LucasXu0**
Draft PR. Only for reference.

### Feature Preview

---

#### PR Checklist

- [ ] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/rust-lib/Cargo.toml [hunk 1]
```diff
@@ -38,9 +39,10 @@ opt-level = 3
 incremental = false
 
 [patch.crates-io]
-collab = { git = "https://github.com/AppFlowy-IO/AppFlowy-Collab"  }
-collab-folder = { git = "https://github.com/AppFlowy-IO/AppFlowy-Collab"  }
-collab-persistence = { git = "https://github.com/AppFlowy-IO/AppFlowy-Collab" }
+collab = { git = "https://github.com/AppFlowy-IO/AppFlowy-Collab", branch = "main"  }
```
Comment (appflowy) [inline frontend/rust-lib/Cargo.toml:42]: the default branch will be the main branch. So no need to mark the branch.

File: frontend/rust-lib/flowy-document2/src/entities.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  // Support customize initial data
+}
+
+#[derive(Default, ProtoBuf)]
+pub struct DocumentPB {
+  #[pb(index = 1)]
+  pub page_id: String,
+
+  #[pb(index = 2)]
+  pub blocks: BlocksPB,
```
Comment (appflowy) [inline frontend/rust-lib/flowy-document2/src/entities.rs:18]: I think BlockMap would be better

File: frontend/rust-lib/flowy-document2/src/document.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    let text_block_id = nanoid!(10);
+    let text_0_children_id = nanoid!(10);
+    let text_block = Block {
+      id: text_block_id.clone(),
+      ty: "text".to_string(),
+      parent: page_id.clone(),
+      children: text_0_children_id.clone(),
+      external_id: None,
+      external_type: None,
+      data: HashMap::new(),
```
Comment (qinluhe) [inline frontend/rust-lib/flowy-document2/src/document.rs:134]: The data will be like this?
<pre>
data: {
    delta: vec![]
}
</pre>

Comment (LucasXu0) [reply frontend/rust-lib/flowy-document2/src/document.rs:134]: Yes.

**Pull request merged**